packages/financial-templates-lib/test/clients/OptimisticOracleEventClient.js

```diff
@@ -0,0 +1,435 @@
+    proposalTime = await optimisticOracle.getCurrentTime();
+    proposalTxn1 = await optimisticOracle.proposePrice(
+      optimisticRequester.address,
+      identifier,
+      requestTime,
+      "0x",
+      correctPrice,
+      {
+        from: proposer
+      }
+    );
+    proposalTxn2 = await optimisticOracle.proposePrice(
+      optimisticRequester.address,
+      identifier,
+      requestTime + 1,
+      "0x",
+      correctPrice,
+      {
+        from: proposer
+      }
+    );
+
+    // Make disputes and resolve them
+    await collateral.approve(optimisticOracle.address, MAX_UINT_VAL, { from: disputer });
+    disputeTxn1 = await optimisticOracle.disputePrice(optimisticRequester.address, identifier, requestTime, "0x", {
+      from: disputer
+    });
+    await pushPrice(correctPrice);
+    disputeTxn2 = await optimisticOracle.disputePrice(optimisticRequester.address, identifier, requestTime + 1, "0x", {
+      from: disputer
+    });
+    await pushPrice(correctPrice);
+
+    // Settle expired proposals and resolved disputes
+    settlementTxn1 = await optimisticOracle.settle(optimisticRequester.address, identifier, requestTime, "0x");
+    settlementTxn2 = await optimisticOracle.settle(optimisticRequester.address, identifier, requestTime + 1, "0x");
+  });
+
+  it("Return RequestPrice events", async function() {
+    await client.clearState();
+    // State is empty before update().
+    assert.deepStrictEqual([], client.getAllRequestPriceEvents());
+    await client.update();
+    assert.deepStrictEqual(client.getAllRequestPriceEvents(), [
+      {
+        transactionHash: requestTxn1.tx,
+        blockNumber: requestTxn1.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: "0x",
+        timestamp: requestTime.toString(),
+        currency: collateral.address,
+        reward: "0",
+        finalFee
+      },
+      {
+        transactionHash: requestTxn2.tx,
+        blockNumber: requestTxn2.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: "0x",
+        timestamp: (requestTime + 1).toString(),
+        currency: collateral.address,
+        reward: "0",
+        finalFee
+      }
+    ]);
+
+    // Correctly adds only new events after last query
+    const newTxn = await optimisticRequester.requestPrice(
+      identifier,
+      requestTime,
+      collateral.address,
+      collateral.address,
+      0
+    );
+    await client.clearState();
+    await client.update();
+    assert.deepStrictEqual(client.getAllRequestPriceEvents(), [
+      {
+        transactionHash: newTxn.tx,
+        blockNumber: newTxn.receipt.blockNumber,
+        requester: optimisticRequester.address,
+        identifier: hexToUtf8(identifier),
+        ancillaryData: collateral.address.toLowerCase(),
```

Comment:
Added comment about this:

```
        // Note: Convert contract address to lowercase to adjust for how Solidity casts addresses to bytes.
        // This is important because `requestPrice` expects `ancillaryData` to be of type bytes,
```

`requestPrice()` expects `ancillaryData` to be `bytes` so passing in `contract.address` here gets casted to bytes.